Cargo.toml

```diff
@@ -23,17 +23,19 @@ color-eyre = "0.6"
 eyre = "0.6"
 flate2 = "1"
 futures = "0.3"
+git2 = "0.17"
 home = "0.5.5"
 human-panic = "1"
 keyring = "2"
-git2 = "0.17"
+protobuf-src = "1.1.0"
```

Comment:
I don't recommend usage of protobuf-src. I used to love it but IMO it's a false economy. It takes a long time to compile and because it has all the dependencies protobuf has, if it goes wrong it has very very hard to debug build issues. I think insofar as you want to ensure a specific protobuf version,it'd be better to check protoc's version deliberately.

Reply:
The compilation errors are catched by prost faik, i would really want to use a vendored thing, to keep the setup reproducible everywhere.

Reply:
What I mean is that protobuf-src runs the protobuf build. I had a bunch of issues where that build was simply failing with some cmake issue, which did not happen if I just compiled protobuf on its own, and this was then very hard to debug. Additionally, you won't be able to rely on protobuf-src for e.g. generating python or ts.

Reply:
Ah! We could also use a vendored version which is precompiled

Reply:
https://crates.io/crates/protoc-bin-vendored

Reply:
Another alternative is to require `protoc` to be present on the host, and check its version range?

Reply:
I would opt against that if somehow possible

Reply:
After having seen the issues caused by `protoc-src` and others, I think that checking for `protoc` and validating its version-range is the lesser-evil here (where you can use the version check to ensure reproducibility). It doesn't "just-work" (which sucks), but I think the other options have much worse worst-cases :(

Reply:
and using the vendored ones? @tetigi @wngr